### PR TITLE
update version to 1.0.1-SNAPSHOT, remove log4j and logback

### LINKED ISSUE #8: log4j-to-slf4j?
You have the following logging dependencies
```xml
        <dependency>
            <groupId>org.apache.logging.log4j</groupId>
            <artifactId>log4j-to-slf4j</artifactId>
            <version>2.11.1</version>
        </dependency>
        
        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-api</artifactId>
            <version>1.7.25</version>
        </dependency>
        
        <dependency>
            <groupId>ch.qos.logback</groupId>
            <artifactId>logback-classic</artifactId>
            <version>1.2.3</version>
        </dependency>
```
While I have these
```xml
		<dependency>
			<groupId>org.apache.logging.log4j</groupId>
			<artifactId>log4j-api</artifactId>
			<version>2.11.1</version>
		</dependency>
		<dependency>
			<groupId>org.apache.logging.log4j</groupId>
			<artifactId>log4j-core</artifactId>
			<version>2.11.1</version>
		</dependency>
		<dependency>
			<groupId>org.apache.logging.log4j</groupId>
			<artifactId>log4j-slf4j-impl</artifactId>
			<version>2.11.1</version>
		</dependency>
		<dependency>
			<groupId>org.apache.logging.log4j</groupId>
			<artifactId>log4j-1.2-api</artifactId>
			<version>2.11.1</version>
		</dependency>
		<dependency>
			<groupId>org.graylog2.log4j2</groupId>
			<artifactId>log4j2-gelf</artifactId>
			<version>1.3.1</version>
		</dependency>
```
And now I'm getting
```java
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.sun.javafx.application.LauncherImpl.launchApplicationWithArgs(LauncherImpl.java:389)
	at com.sun.javafx.application.LauncherImpl.launchApplication(LauncherImpl.java:328)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:767)
Caused by: java.lang.RuntimeException: Exception in Application start method
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:917)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$154(LauncherImpl.java:182)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.StackOverflowError
	at java.lang.reflect.InvocationTargetException.<init>(InvocationTargetException.java:72)
	at sun.reflect.GeneratedMethodAccessor1.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.logging.log4j.util.StackLocator.getCallerClass(StackLocator.java:110)
	at org.apache.logging.log4j.util.StackLocator.getCallerClass(StackLocator.java:123)
	at org.apache.logging.log4j.util.StackLocatorUtil.getCallerClass(StackLocatorUtil.java:55)
	at org.apache.logging.slf4j.Log4jLoggerFactory.getContext(Log4jLoggerFactory.java:42)
	at org.apache.logging.log4j.spi.AbstractLoggerAdapter.getLogger(AbstractLoggerAdapter.java:46)
	at org.apache.logging.slf4j.Log4jLoggerFactory.getLogger(Log4jLoggerFactory.java:29)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:358)
	at org.apache.logging.slf4j.SLF4JLoggerContext.getLogger(SLF4JLoggerContext.java:39)
	at org.apache.logging.slf4j.Log4jLoggerFactory.newLogger(Log4jLoggerFactory.java:37)
	at org.apache.logging.slf4j.Log4jLoggerFactory.newLogger(Log4jLoggerFactory.java:29)
	at org.apache.logging.log4j.spi.AbstractLoggerAdapter.getLogger(AbstractLoggerAdapter.java:52)
	at org.apache.logging.slf4j.Log4jLoggerFactory.getLogger(Log4jLoggerFactory.java:29)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:358)
	( etc )
```

What's happening (I think) is that in `Workbench` you ask log4j for a new logger. `log4j-to-slf4j` will pass this request on to slf4j, but my `log4j-slf4j-impl` gives it back to log4j, resulting in an endless loop.

Considering that you're the library, you should remove log4j-to-slf4j as well as the logback dependency. A full explanation is [here](https://www.baeldung.com/slf4j-with-log4j2-logback), but the TL;DR is that because everyone has their own favorite logging framework (log4j, logback, java.util.logging) SLF4J was made to unify all the different logging frameworks. Library creators should use SLF4J for their logging, library users should use the SLF4J implementation they want to use. Which in my case is Log4j2.

### PR DESCRIPTION
**Pull request by sirolf2009**
Fixes #8

### CODE REVIEW

**Comment by iajn** [comment]
I don't think you fully understand what slf4j is and does. Its a log facade, basically just an API, that bridges logging to implementations (log4j, log4j2, logback). This way, users of your library are not forced to use the same logging implementation (which was the cause of the original issue).

To answer your points:
1. There is no "slf4j configuration". The log4j2.xml is no longer needed. Configuring the logger falls to the user of your library, and they configure it how they see fit for their application.

2. The output is caused by missing slf4j implementation. Adding one as a **test dependency** fixes the issue.

**Comment by sirolf2009** [comment]
sorry, I had completely missed your reply.
iajn is right, I've removed src/main/resources/pom.xml because it's unnecessary now that there isn't a log4j anymore. There's also nothing to port, because a library shouldn't come with its own logging config, the user should create his own. I've left src/test/resources/pom.xml and added log4j-slf4j-impl as a test dependency. So now when the tests are ran, it'll use a log4j implementation and not complain op NOP loggers.

File: workbenchfx-core/pom.xml [hunk 1]
```diff
@@ -304,6 +304,17 @@
                         <goals>
                             <goal>shade</goal>
                         </goals>
+                        <configuration>
```
Comment (martinfrancois) [inline workbenchfx-core/pom.xml]: What is this change related to?

**Comment by martinfrancois** [comment]
Since there was no answer from @sirolf2009, I reverted the unwanted change, now it's good to merge.
